File: packages/ios-swift/Sources/CinacoinAppKitAuth/Models/AuthResult.swift
import Foundation

/// Authentication result
public struct AuthResult: Equatable, Sendable {
    public let userId: String
    public let email: String?
    public let provider: AuthProvider?
    public let accessToken: String
    public let refreshToken: String?
    public let expiresAt: Date
    
    public init(
        userId: String,
        email: String? = nil,
        provider: AuthProvider? = nil,
        accessToken: String,
        refreshToken: String? = nil,
        expiresAt: Date
    ) {
        self.userId = userId
        self.email = email
        self.provider = provider
        self.accessToken = accessToken
        self.refreshToken = refreshToken
        self.expiresAt = expiresAt
    }
    
    /// Check if token is expired
    public var isExpired: Bool {
        Date() >= expiresAt
    }
}

/// Supported auth providers
public enum AuthProvider: String, CaseIterable, Sendable {
    case google
    case apple
    case github
    case discord
    case email
}

File: packages/android-kotlin/src/main/kotlin/com/cinacoin/appkit/auth/model/AuthResult.kt
package com.cinacoin.appkit.auth.model

import java.util.Date

data class AuthResult(
    val userId: String,
    val email: String? = null,
    val provider: AuthProvider? = null,
    val accessToken: String,
    val refreshToken: String? = null,
    val expiresAt: Long
) {
    val isExpired: Boolean
        get() = System.currentTimeMillis() >= expiresAt
}

enum class AuthProvider(val value: String) {
    GOOGLE("google"),
    APPLE("apple"),
    GITHUB("github"),
    DISCORD("discord"),
    EMAIL("email")
}
